### ISSUE TITLE
Can't run with Swift upload outside of main thread

### ISSUE THREAD

**Issue by MartinPaulo** [gvlproject/python-genomespaceclient#2]
The `SwiftStorageHandler` performs an upload by calling the OpenStack `swiftclient.shell.main` method. But this `main` method has the following line it: `signal.signal(signal.SIGINT, immediate_exit)`. 

The net effect is that if you try to do an upload to Swift outside of the main thread (e.g.: in a web application) an exception will be thrown with the following message: `ValueError: signal only works in main thread`.

So as written there is a broad category of applications in which this this code is not usable :(

**Comment by nuwang** [comment]
Thanks for reporting this Martin. So much for taking shortcuts with swiftclient.shell.main :-) There's also a system.exit() further on, so I wonder why things didn't fail more catastrophically during tests for example?

As I recall, the reason this was done originally was because the python-swiftclient library doesn't offer a parallelised upload method, all of that logic resides within swiftclient.shell, so the only way to reuse it was through swiftclient.shell.main (It is not particularly well-factored).

This will all have to be redone now. I'll try to get onto this asap.

**Comment by MartinPaulo** [comment]
My workaround at the moment is:

```python
cmd = 'genomespace -t %s cp %s %s' % (token, file, bucket)
(out, err) = Popen(cmd, stdin=PIPE, stdout=PIPE,
                               stderr=PIPE, close_fds=True,
                               shell=True).communicate(None)
```

Not ideal, but it does move the files!

Thanks for the quick response

**Comment by nuwang** [comment]
@MartinPaulo Sorry this took a bit longer than I expected - have switched dependencies to cloudbridge but that required some changes to cloudbridge + a new release of cloudbridge. 
Now fixed in patch:
https://github.com/gvlproject/python-genomespaceclient/commit/c62832b12d89e82e5173defcf048feb8642ee95b

I was thinking of holding off on a new release for a few more days to get some more enhancements in (mainly, support for copying multiple files and folders). If that's ok, you can use the dev version of genomespaceclient with pip install git+git://github.com/gvlproject/python-genomespaceclient. Otherwise, I can expedite the release. Let me know.

**Comment by nuwang** [comment]
@MartinPaulo A new release has been pushed to pypi - let me know whether it works ok. Have also added support for copying folders recursively and wildcard copying.

**Comment by MartinPaulo** [comment]
It works once I apply a local fix for issue #4
